src/table/components/head/HeadCellMenu.tsx

```diff
@@ -1,110 +1,142 @@
-import React, { useRef, useState } from 'react';
+import React, { useRef, useState, useMemo } from 'react';
 import ClickAwayListener from '@mui/material/ClickAwayListener';
 import Grow from '@mui/material/Grow';
 import Paper from '@mui/material/Paper';
 import Popper from '@mui/material/Popper';
 import MenuList from '@mui/material/MenuList';
-import ListItem from '@mui/material/ListItem';
-import ListItemButton from '@mui/material/ListItemButton';
-import ListItemIcon from '@mui/material/ListItemIcon';
-import ListItemText from '@mui/material/ListItemText';
 import ArrowUpwardIcon from '@mui/icons-material/ArrowUpward';
 import ArrowDownwardIcon from '@mui/icons-material/ArrowDownward';
+import SearchIcon from '@mui/icons-material/Search';
 import MoreHoriz from '@mui/icons-material/MoreHoriz';
-import { StyledMenuIconButton, StyledCellMenu } from './styles';
-import { HeadCellMenuProps } from '../../types';
-import { SortDirection } from '../../../types';
+import { HeadCellMenuProps, HeadCellMenuGroup } from '../../types';
+import useListboxFilter from '../../hooks/use-listbox-filter';
+import MenuGroup from './MenuGroup';
+import { StyledMenuIconButton, StyledCellMenu, NebulaListBox, PrimaryDropdownPaper } from './styles';
 
 export default function HeadCellMenu({
   headerStyle,
   translator,
   sortDirection,
   sortFromMenu,
+  embed,
+  layout,
+  columnIndex,
   isInteractionEnabled,
   isCurrentColumnActive,
 }: HeadCellMenuProps) {
-  const [open, setOpen] = useState(false);
+  const [openPrimaryDropdown, setOpenPrimaryDropdown] = useState(false);
+  const [openSecondaryDropdown, setOpenSecondaryDropdown] = useState(false);
   const anchorRef = useRef<HTMLButtonElement>(null);
+  const elRef = useRef<HTMLElement | undefined>();
+  useListboxFilter({ elRef, layout, embed, columnIndex, openPrimaryDropdown });
 
-  const MenuItem = ({
-    children,
-    itemTitle,
-    newSortDirection,
-  }: {
-    children: any;
-    itemTitle: string;
-    newSortDirection: SortDirection;
-  }) => {
-    const isDisabled = !isInteractionEnabled || (isCurrentColumnActive && newSortDirection === sortDirection);
-    return (
-      <ListItem disablePadding>
-        <ListItemButton
-          disabled={isDisabled}
-          className="sn-table-head-menu-item-button"
-          onClick={(evt) => {
-            sortFromMenu(evt, newSortDirection);
-            setOpen(false);
-          }}
-        >
-          <ListItemIcon sx={{ minWidth: '25px' }}>{children}</ListItemIcon>
-          <ListItemText primary={itemTitle} />
-        </ListItemButton>
-      </ListItem>
-    );
-  };
+  const getMenuItems = useMemo<HeadCellMenuGroup[]>(
+    () => [
+      {
+        id: 1,
+        menus: [
```

Comment:
These are group of menus, i had a bigger scale for this use case regarding submenus in multiple level (recursively iterating over this object and rendering `<MenuGroup />` component for each subset) but right now since i removed that part, the only use case for this is for determining when to show the group divider component. 

btw, do you mean changing the `menus` to `options`?